RecoTracker/MkFitCore/interface/IdxChi2List.h

```diff
@@ -0,0 +1,35 @@
+#ifndef RecoTracker_MkFitCore_interface_IdxChi2List_h
+#define RecoTracker_MkFitCore_interface_IdxChi2List_h
+
+namespace mkfit {
+
+  struct IdxChi2List {
+  public:
+    unsigned int module;  // module id
+    int hitIdx;           // hit index
+    int trkIdx;           // candidate index
+    int nhits;            // number of hits (used for sorting)
+    int ntailholes;       // number of holes at the end of the track (used for sorting)
+    int noverlaps;        // number of overlaps (used for sorting)
+    int nholes;           // number of holes (used for sorting)
+    float pt;             // pt (used for sorting)
+    float chi2;           // total chi2 (used for sorting)
+    float chi2_hit;       // chi2 of the added hit
+    float score;          // score used for candidate ranking
+
+    IdxChi2List() = default;
+    IdxChi2List& operator=(const IdxChi2List&) = default;
+
+    IdxChi2List(unsigned int) { reset(); }
```

Comment:
the `int` argument is confusing.
Isn't it more clear to just add default initialization above ` float score = 0;` ?

Reply:
During tracking we create the struct on the stack and then manually assign to every member.

There was a second usage when dumping hit-match-info to a TTree for NN cand-score training. There, in creation of the exported structure in selectHitIndicies() we do not know yet if the given hit will be selected -- so we pre-initialize the struct to all zeroes there (as we might not get to process it later).

I have committed a comment above the unsigned int constructor to clarify this.